File: array_string/_go/pascal_triangle.go
package main

import "fmt"

func getRow(A int) []int {
	var arr [][]int
	for i := 1; i <= A+1; i++ {
		temp := make([]int, i)
		arr = append(arr, temp)
	}
	for i := 0; i < len(arr); i++ {
		arr[i][0] = 1
		arr[i][len(arr[i])-1] = 1
	}

	for row := 2; row < len(arr); row++ {
		i := 1
		for x := 1; x < len(arr[row-1]); x++ {
			arr[row][i] = arr[row-1][x-1] + arr[row-1][x]
			i++
		}
	}
	return arr[A]
}

func main() {
	fmt.Println(getRow(3))
	//arr := []int{1, 2, 3, 4, 5}
	//fmt.Println(arr[len(arr)-1])
}

File: array_string/pascal_triangle.py
def getRow(A):
    arr = [[0] * (i + 1) for i in range(A + 1)]
    for i in range(len(arr)):
        arr[i][0] = 1
        arr[i][-1] = 1

    for row in range(2, len(arr)):
        i = 1
        for val in range(1, len(arr[row - 1])):
            arr[row][i] = arr[row - 1][val - 1] + arr[row - 1][val]
            i += 1
    return arr[A]


print(getRow(3))


